app/src/main/kotlin/com/simplemobiletools/calculator/activities/DrawActivity.java

```diff
@@ -60,6 +69,25 @@ public void supersedeSketchSheetView(SketchSheetView equationSketchView) {
         this.equationSketchView = equationSketchView;
     }
 
+    public void recoverAfterError(String error,String errorMessage){
+        findViewById(R.id.loading).setVisibility(View.GONE);
+        AlertDialog alertDialog = new AlertDialog.Builder(DrawActivity.this,R.style.MyDialogTheme).create();
+        alertDialog.setTitle(error);
+        alertDialog.setMessage(errorMessage);
+        alertDialog.setButton(AlertDialog.BUTTON_NEGATIVE, "OK",
+                new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.dismiss();
+                    }
+                });
+        alertDialog.show();
+    }
+
+    public void supersedeBuilder(BackgroundGetGraphTaskBuilder builder) {
```

Comment:
add another `@TestOnly`. I'm bikeshedding 🚲🏠 a bit, but we should make it obvious that it's for testing only and should never be used for prod purpose.

Reply:
🚲🏠 Bikesheds are cool. Done